### PR TITLE
Task-45890: Update scheduling drawer

### PR DESCRIPTION
**Pull request by hassenelkamel-exo**
In this PR we will:

- Update scheduling drawer permissions and dateTime picker rules
- Implement edit scheduling
- Implement cancel scheduling

### CODE REVIEW

File: services/src/main/java/org/exoplatform/news/rest/NewsRestResourcesV1.java [hunk 1]
```diff
@@ -916,6 +949,13 @@ private boolean canViewNews(String authenticatedUser, Space space) throws Except
     return spaceService.isSuperManager(authenticatedUser) || (space != null && spaceService.isMember(space, authenticatedUser));
   }
 
+  private boolean canScheduleNews(String authenticatedUser, Space space) {
+    boolean isManager = space != null && spaceService.isManager(space, authenticatedUser);
+    boolean isRedactor = space != null && spaceService.isRedactor(space, authenticatedUser);
+    boolean spaceHasARedactor = space != null && space.getRedactors() != null && space.getRedactors().length > 0;
+    return ((spaceHasARedactor && (isRedactor || isManager)) || (!spaceHasARedactor && isManager));
```
Comment (azayati) [inline services/src/main/java/org/exoplatform/news/rest/NewsRestResourcesV1.java]:
```suggestion
    return spaceService.isManager(space, authenticatedUser) || spaceService.isRedactor(space, authenticatedUser);
```

File: services/src/main/resources/locale/portlet/news/News_en.properties [hunk 2]
```diff
@@ -24,14 +24,20 @@ news.composer.draft.savingDraftStatus=Saving draft...
 news.composer.draft.savedDraftStatus=Draft saved
 news.composer.draft.delete.confirmation=Are you sure you want to delete this draft?
 news.composer.postArticle=Post an Article
+news.composer.editArticle=Edit Scheduling
 news.composer.postImmediately=Post immediately
 news.composer.postLater=Post later
+news.composer.notPost=Dont't post
 news.composer.choosePostDate=Choose when you want your article to be posted:
+news.composer.chooseNotPost=Choose when you no longer want to publish your article.
```
Comment (azayati) [inline services/src/main/resources/locale/portlet/news/News_en.properties]:
```suggestion
news.composer.chooseNotPost=Choose when you no longer want to post your article.
```

File: webapp/src/main/webapp/schedule-news/components/ExoScheduleNewsDrawer.vue [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+      const newDate = new Date(this.postDate);
+      newDate.setHours(this.postDateTime.getHours());
+      newDate.setMinutes(this.postDateTime.getMinutes());
+      newDate.setSeconds(0);
+      this.postDate = newDate;
+    },
+  },
+  computed: {
+    saveButtonLabel() {
+      return this.postArticleMode==='later' ? this.$t('news.composer.schedule'): this.$t('news.composer.post');
```
Comment (azayati) [inline webapp/src/main/webapp/schedule-news/components/ExoScheduleNewsDrawer.vue]:
```suggestion
      return this.postArticleMode === 'later' ? this.$t('news.composer.schedule'): this.$t('news.composer.post');
```

File: webapp/src/main/webapp/schedule-news/components/ExoScheduleNewsDrawer.vue [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+    schedulePostDate: null,
+    postDate: null,
+  }),
+  watch: {
+    postDate(newVal, oldVal) {
+      if (!this.postDate || !newVal || !oldVal || new Date(newVal).getTime() === new Date(oldVal).getTime()) {
+        return;
+      }
+      if ( (new Date(newVal).getTime() !== new Date(oldVal).getTime()) || (this.postDate === this.schedulePostDate)){
+        this.disabled = false;
```
Comment (azayati) [inline webapp/src/main/webapp/schedule-news/components/ExoScheduleNewsDrawer.vue]: Is the disable considered ?

File: webapp/src/main/webapp/schedule-news/main.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+// get overrided components if exists
+if (extensionRegistry) {
+  const components = extensionRegistry.loadComponents('ScheduleNewsDrawer');
+  if (components && components.length > 0) {
+    components.forEach(cmp => {
+      Vue.component(cmp.componentName, cmp.componentOptions);
+    });
+  }
+}
+let shareNewsApp;
```
Comment (azayati) [inline webapp/src/main/webapp/schedule-news/main.js]:
```suggestion
let scheduleNewsDrawer
```
?

File: webapp/webpack.common.js [hunk 6]
```diff
@@ -15,6 +15,7 @@ let config = {
     newsSearchCard: './src/main/webapp/news-search/main.js',
     newsExtensions: './src/main/webapp/news-extensions/main.js',
     newsDetailsApp: './src/main/webapp/news-details-app/main.js',
+    scheduleNewsDrawer: './src/main/webapp/schedule-news/main.js',
```
Comment (azayati) [inline webapp/webpack.common.js]:
```suggestion
    scheduleNewsDrawer: './src/main/webapp/schedule-news-drawer/main.js',
```

**Review by azayati — CHANGES_REQUESTED** [review]
See review

**Review by azayati — DISMISSED** [review]
LGTM after QA validation

File: services/src/main/java/org/exoplatform/news/rest/NewsRestResourcesV1.java [hunk 7]
```diff
@@ -748,7 +748,7 @@ public Response undoDeleteNews(
   @Produces(MediaType.TEXT_PLAIN)
   @RolesAllowed("users")
   @ApiOperation(value = "check if the current user can create a news in the given space", httpMethod = "GET", response = Response.class, notes = "This checks if the current user can create a news in the given space", consumes = "application/json")
-  @ApiResponses(value = { @ApiResponse(code = 200, message = "News deleted"),
+  @ApiResponses(value = { @ApiResponse(code = 200, message = "News created"),
```
Comment (ahamdi) [inline services/src/main/java/org/exoplatform/news/rest/NewsRestResourcesV1.java]:
```suggestion
  @ApiResponses(value = { @ApiResponse(code = 200, message = "User ability to create a news is returned"),
```

File: services/src/main/java/org/exoplatform/news/rest/NewsRestResourcesV1.java [hunk 8]
```diff
@@ -775,6 +775,39 @@ public Response canCreateNews(@Context HttpServletRequest request,
       return Response.serverError().build();
     }
   }
+
+  @GET
+  @Path("canScheduleNews/{spaceId}")
+  @Produces(MediaType.TEXT_PLAIN)
+  @RolesAllowed("users")
+  @ApiOperation(value = "check if the current user can schedule a news in the given space", httpMethod = "GET", response = Response.class, notes = "This checks if the current user can schedule a news in the given space", consumes = "application/json")
+  @ApiResponses(value = { @ApiResponse(code = 200, message = "News scheduled"),
```
Comment (ahamdi) [inline services/src/main/java/org/exoplatform/news/rest/NewsRestResourcesV1.java]:
```suggestion
  @ApiResponses(value = { @ApiResponse(code = 200, message = "User ability to schedule a news"),
```

File: webapp/src/main/webapp/schedule-news-drawer/components/ExoScheduleNewsDrawer.vue [hunk 9]
```diff
@@ -0,0 +1,211 @@
...
+      if (!this.postDate || !newVal || !oldVal || new Date(newVal).getTime() === new Date(oldVal).getTime()) {
+        return;
+      }
+      const postDate = new Date(this.postDate);
+      const scheduleDate = new Date(this.schedulePostDate);
+      if (postDate.getTime() !== scheduleDate.getTime()){
+        this.disabled = false;
+      } else {
+        this.disabled = true;
+      }
```
Comment (ahamdi) [inline webapp/src/main/webapp/schedule-news-drawer/components/ExoScheduleNewsDrawer.vue]: Could be simplified by
```suggestion
        this.disabled = postDate.getTime() === scheduleDate.getTime();
```

**Review by ahamdi — DISMISSED** [review]

**Review by ahamdi — APPROVED** [review]

**Pull request merged**